### PR TITLE
Improved docs for std.file.slurp

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

**Comment by JackStouffer** [comment]
Interesting thing to note about this function: it's impossible to allow the filename to be an input range with the current design.

**Comment by veelo** [comment]
I'd end every line with a full-stop, in all "Improved docs" PRs.

**Comment by JackStouffer** [comment]
> I'd end every line with a full-stop

Not sure what you mean

**Comment by veelo** [comment]
> > I'd end every line with a full-stop.

> Not sure what you mean

I mean like, in this PR:

`... be returned as.`
`... to read.`
`... when reading.`
`... is empty.`

There are similar cases (missing full stop, period, point, dot) in your other PRs :-)

**Comment by JackStouffer** [comment]
That's intentional. There are no periods in the Params or other sections as each of the sections are not complete sentences

**Comment by veelo** [comment]
> That's intentional.

Fair enough. There are places in the docs where parameters are formatted with capitals and full stops (http://dlang.org/phobos/std_algorithm_searching.html#find) which look good on my eyes though.

**Comment by wilzbach** [comment]
> Fair enough. There are places in the docs where parameters are formatted with capitals and full stops 

Maybe that's something simple where we should just agree upon?
In case someone is interested in number:

```
> grep -ri params: * | wc -l
955
```

Number of arguments

```
> grep -A 20 -ri params: * | grep "=.*[^;]$" | wc -l
2619
```

Arguments with dot (not multi-line)

```
> grep -A 20 -ri params: * | grep "=.*[^;]$" | grep "=.*[.]$" | less | wc -l
843
```

Disclaimer: I prefer @JackStouffer's version without dots for params.

File: std/file.d [hunk 1]
```diff
@@ -4005,7 +4005,23 @@ unittest
...
+ *     Types = the types that each of the elements in the line should be returned as
+ *     filename = the name of the file to read
+ *     format = the format string to use when reading
+ *
+ * Returns:
+ *     If only one type is passed, then an array of that type. Otherwise, an
+ *     array of $(REF Tuple, std,typecons)s.
+ *
+ * Throws:
+ *     `Exception` if one of the lines is empty
```
Comment (qznc) [inline std/file.d]: It also throws if there are trailing characters after the format string. At least, I can see an `enforce` down there.

Also, formattedRead has an enforce about the format string, which might throw. This is not documented as well. :disappointed:

**Comment by JackStouffer** [comment]
@qznc Fixed

**Comment by qznc** [comment]
LGTM :smile:

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Pull request merged**